### PR TITLE
Adding alerts worker to check for alerts

### PR DESCRIPTION
**Pull request by nimrodshn**
~***This is a work in progress!***~

Adding the alert worker to check for alerts periodically.

cc : @yaacov

### CODE REVIEW

File: alerts/utils.go [hunk 1]
```diff
@@ -29,6 +34,32 @@ func (alert *Alert) setStatus(status bool){
...
+		fmt.Printf("maintenance: start\n")
+		a.checkAlerts()
+	}
+}
+
+func (a *Alerts) checkAlerts() {
+	for _, alert_list := range a.AlertLists {
+		// Get each tenants item list
+		for _, alert := range alert_list.List {
+			raw_data := a.Backend.GetRawData(alert_list.Tenant.Id ,alert.Id,80,0, 0,"ASC")
```
Comment (nimrodshn) [inline alerts/utils.go]: I am not sure what method to call here to get the last value / last X values.
cc: @yaacov

**Comment by nimrodshn** [comment]
@yaacov once this is working I would like to add tests to this PR to check the entire functionality of alerts.

**Comment by yaacov** [comment]
ahh... almost missed the wip thingy ... good that you added the !!! :-)

File: alerts/utils.go (hunk 1 above)
Comment (yaacov) [reply alerts/utils.go]: if it works it is good enough for now, re-factor after you tested it and you think you know what needs to be re-factored and what is not.

Comment (nimrodshn) [reply alerts/utils.go]: @yaacov I'm not sure what values I should put at `start` & `end` for the `GetRawData`??

Comment (yaacov) [reply alerts/utils.go]: https://github.com/MohawkTSDB/mohawk/blob/master/backend/backend.go#L59

`start` and `end` are javascript timestamp [1] in milisec 
limit is the number of samples you want
order is DESC of ASC string

like an sql query:
`select from reading when time > start and time < and sort by time ASC limit 1`

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

Comment (nimrodshn) [reply alerts/utils.go]: @yaacov hmm but I dont care about `end` and `start` and timestamps.. I just want the last time/value pair inserted. Isn't there a simple method for getting this data?

Comment (yaacov) [reply alerts/utils.go]: you can try `GetItemList` but it has other drawbacks.
or you can implement a new callback if you want in all backends ...

**Comment by nimrodshn** [comment]
**Still not working**

I have an issue with the following line:
https://github.com/MohawkTSDB/mohawk/blob/abb3678fe3db2f27b52db425da671a4eb5db75a9/storage/memory/memory.go#L148

When we run `Open` with an `Alerts` object this fills the array with zero object...

**Comment by nimrodshn** [comment]
Working! found the bug! just need to write unit tests and its ready to go in :)

**Comment by nimrodshn** [comment]
Current yaml file should look like this:
```
backend: "memory"
port: 8085
alerts:
- id: "free_memory is low"
  metric: "free_memory"
  from: 2000
  to: 2100
  type: 0
- id: "free_memory is extremly low"
  metric: "free_memory"
  to: 1500
  type: 2
```

**Comment by yaacov** [comment]
a. changed some files ^^ , you should rebase.
b. add the config-example.yaml file someware, it can be helpful.

**Comment by yaacov** [comment]
> just need to write unit tests and its ready to go in :)

great ! next PR, merging now

**Comment by yaacov** [comment]
p.s.

next after unit tests, is making it send alert status change to an alert buffer [1]

[1] https://github.com/ilackarms/message-buffer

**Pull request merged**